src/main/java/net/minecraftforge/fluids/BlockFluidBase.java

```diff
@@ -802,4 +802,10 @@ private boolean isWithinFluid(IBlockAccess world, BlockPos pos, Vec3d vec)
         return filled < 0 ? vec.y > pos.getY() + filled + 1
                           : vec.y < pos.getY() + filled;
     }
+    
+    @Override
+    public float getBlockLiquidHeight(World world, BlockPos pos, IBlockState state, Material material)
+    {
+        return getFilledPercentage(world, pos);
```

Comment:
What are you planning to do here for negative-density fluids?

Specifically:
https://github.com/MinecraftForge/MinecraftForge/blob/857573f10fbb18b3b15d7053757a2216730b459d/src/main/java/net/minecraftforge/fluids/IFluidBlock.java#L79-L80

Reply:
Oh right, maybe return zero?